### PR TITLE
Feature: Add ability to specify jest presets

### PR DESCRIPTION
**Pull request by robert3005**
By default configures ts-jest preset to support writing tests in typescript. You need to provide `typescript` and `ts-jest` dependencies on testDeps configuration to run typescript tests.

### CODE REVIEW

File: gradle-typescript/src/main/java/com/gradlets/gradle/typescript/JestTestSupport.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+public final class JestTestSupport {
+
+    private JestTestSupport() {}
+
+    static void addDependencyConstraints(Project project, SourceSet testSourceSet) {
+        project.getConfigurations()
+                .named(testSourceSet.getCompileConfigurationName())
+                .configure(conf -> {
+                    conf.getDependencyConstraints()
+                            .add(project.getDependencies().getConstraints().create("npm:types/node:14.6.2"));
```
Comment (ferozco) [inline gradle-typescript/src/main/java/com/gradlets/gradle/typescript/JestTestSupport.java]: can we pull out all of the coordinates into a set and save on the boilerplate?

Comment (robert3005) [reply gradle-typescript/src/main/java/com/gradlets/gradle/typescript/JestTestSupport.java]: we can

File: gradle-typescript/src/main/java/com/gradlets/gradle/typescript/JestTestSupport.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.Set;
+import org.gradle.api.Project;
+
+public final class JestTestSupport {
+
+    private JestTestSupport() {}
+
+    static void addDependencyConstraints(Project project, SourceSet testSourceSet) {
+        Set<String> coordinatesToConstraint = Set.of(
```
Comment (ferozco) [inline gradle-typescript/src/main/java/com/gradlets/gradle/typescript/JestTestSupport.java]: make these static?

**Comment by ferozco** [comment]
a nit but other looks good

**Review by ferozco — APPROVED** [review]

**Pull request merged**